types/logfmt/logfmt-tests.ts

```diff
@@ -0,0 +1,117 @@
+import * as logfmt from 'logfmt';
+import * as through from 'through';
+import * as http from 'http';
+import * as express from 'express';
+
+// Examples taken from project README
+
+logfmt.stringify({foo: 'bar'});
+logfmt.parse('foo=bar');
+
+const logfmt2 = new logfmt();
+logfmt2.stringify = JSON.stringify;
+logfmt2.log({foo: 'bar'});
+logfmt.log({foo: 'bar'});
+
+logfmt.stringify({foo: "bar", a: 14, baz: 'hello kitty'});
+logfmt.parse("foo=bar a=14 baz=\"hello kitty\" cool%story=bro f %^asdf code=H12");
+
+process.stdin.pipe(logfmt.streamParser());
+
+process.stdin
+  .pipe(logfmt.streamStringify())
+  .pipe(process.stdout);
+
+process.stdin
+  .pipe(logfmt.streamParser())
+  .pipe(through((object) => {
+    console.log(object);
+  }));
+
+http.createServer((req, res) => {
+  req.pipe(logfmt.streamParser())
+      .pipe(through((object) => {
+        console.log(object);
+      }));
+
+  res.writeHead(200, {'Content-Type': 'text/plain'});
+  res.end('OK');
+}).listen(3000);
+
+const app = express();
+app.use(logfmt.bodyParserStream());
+app.post('/logs', (req, res) => {
+  if (!req.body) return res.send('OK');
+
+  req.body.pipe(through((line) => {
+    console.dir(line);
+  }));
+
+  res.send('OK');
+});
+
+const app2 = express();
+app2.use(logfmt.bodyParser());
+// req.body is now an array of objects
+app2.post('/logs', (req, res) => {
+  console.log('BODY: ' + JSON.stringify(req.body));
+  req.body.forEach((data: object) => {
+    console.log(data);
+  });
+  res.send('OK');
+});
+
+logfmt.log({foo: "bar", a: 14, baz: 'hello kitty'});
+logfmt.log({foo: "bar", a: 14, baz: 'hello kitty'});
+logfmt.log({foo: "bar", a: 14, baz: 'hello kitty'}, process.stderr);
+
+// logfmt's trick of copying the prototype onto itself seems to trip up
+// TypeScript on module-level member assignment, so this should work but doesn't
+//
+// $ExpectError
+logfmt.stream = process.stderr;
```

Comment:
FTR, this is the one potential issue I'm aware of with these definitions. I couldn't figure out how to make the module-level `stream` (or `maxErrorLines`) variable writable without duplicating the class definition. Open to suggestions.